theano/sandbox/mkl/__init__.py

```diff
@@ -0,0 +1,254 @@
+        name = (kwargs and kwargs.pop('name')) or local_opt.__name__
+        mkl_optimizer.register(name, local_opt, 'fast_run', 'fast_compile',
+                               'mkl', *tags, **kwargs)
+        return local_opt
+    return f
+
+
+class MKLVersion(gof.Op):
+    def c_header_dirs(self):
+        return ldflags(libs=False, include_dir=True)
+
+    def c_libraries(self):
+        return ldflags()
+
+    def c_lib_dirs(self):
+        return ldflags(libs=False, libs_dir=True)
+
+    def make_node(self):
+        return gof.Apply(self, [], [gof.Generic()()])
+
+    def c_support_code(self):
+        ccode = header_text()
+        ccode += """
+        #if PY_MAJOR_VERSION >= 3
+            #define PyInt_FromLong PyLong_FromLong
+        #endif
+        """
+        return ccode
+
+    def c_code(self, node, name, inputs, outputs, sub):
+        o = outputs[0]
+        return textwrap.dedent(
+            """
+            MKLVersion v;
+            mkl_get_version(&v);
+            %(o)s = PyInt_FromLong(atoi(v.Build));
+            """) % locals()
+
+    def c_code_cache_version(self):
+        # Not needed, but make it clear that we do not want to cache this.
+        return None
+
+
+def mkl_version():
+    """
+    Return the current mkl version (e.g., 20160701) we compile with.
+    While `0` indicate fail to get mkl version.
+    """
+    if mkl_version.v is None:
+        try:
+            f = theano.function([], MKLVersion()(),
+                                theano.Mode(optimizer=None),
+                                profile=False)
+            mkl_version.v = f()
+        except:
+            mkl_version.v = 0
+    return mkl_version.v
+
+mkl_version.v = None
+
+
+def _dnn_is_able_to_link():
+    if _dnn_is_able_to_link.status is None:
+        preambule = """
+            #include <stdio.h>
+            #include <mkl_dnn.h>
+            """
+        preambule += textwrap.dedent(header_text())
+
+        body = textwrap.dedent(
+            """
+            dnnError_t err;
+            dnnLayout_t usr_layout = NULL;
+            size_t size[1] = {256};
+            size_t stride[1] = {1};
+
+            if ((err = dnnLayoutCreate_F32(&usr_layout, 1, size, stride)) != E_SUCCESS) {
+                fprintf(stderr, "Failed to create user layout with mkl: %s", err);
+                return (-1);
+            }
+            """)
+
+        if 'mklml_intel' in config.blas.ldflags:
+            params = ['-l', 'mklml_intel', '-l', 'iomp5']
+        elif 'mklml_gnu' in config.blas.ldflags:
+            params = ['-l', 'mklml_gnu', '-l', 'iomp5']
+        elif 'mkl_rt' in config.blas.ldflags:
+            params = ['-l', 'mkl_rt']
+        else:
+            params = []
+
+        compilation_ok, run_ok, out, err = Compiler._try_flags(
+            flag_list=params, preambule=preambule, body=body,
+            try_run=True, output=True, compiler=theano.config.cxx,
+            comp_args=False)
+
+        mkl_available.msg = (
```

Comment:
I think you should set this message only if compilation_ok isn't True or run_ok isn't True